.github/actions/java/action.yml

```diff
@@ -0,0 +1,9 @@
+name: "Set up Adopt OpenJDK 17"
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/setup-java@v3
```

Comment:
Any reason we're not using v4? That's what we are using in Pegasus, and v3 is giving these warnings:

<img width="909" alt="image" src="https://github.com/Airthings/KmpLog/assets/270292/6099db76-88c9-4ed6-841a-6991dc44abc6">

```suggestion
    - uses: actions/setup-java@v4
```